### ISSUE TITLE
Deploying an AppGroup with the same name after delete fails the new AppGroup

### ISSUE THREAD

**Issue by jonathan-innis** [Azure/orkestra#240]
**Describe the bug**

**To Reproduce**
1. Create an ApplicationGroup (i.e. the sample bookinfo ApplicationGroup)
2. Delete the ApplicationGroup
3. While the ApplicationGroup is in a deleting state, re-create the ApplicationGroup

This should cause the ApplicationGroup to show the following error and never recover out of this state
```
NAME       DEPLOY      READY    MESSAGE                                            AGE
bookinfo   Succeeded   Failed   workflow in failure/error condition : %!w(<nil>)   3m20s
```

This is particularly noticeable in testing as re-creating the same AppGroup is a pretty common action 

**Expected behavior**
ApplicationGroup should finish deletion and then pick-up the new ApplicationGroup and start deploying again

**Comment by jonathan-innis** [comment]
Assigning to myself because this is somewhat blocking the testing of the ApplicationGroups

**Comment by jonathan-innis** [comment]
I suspect I know what's going on here. The namespace is in a deleting state until all of the items in the namespace are gone. When the executor tries to re-create, it fails because the namespace is still deleting. Trying to remedy this with retry in the executor

**Comment by nitishm** [comment]
Ah yes that makes sense. That said, how can we make debugging easier? A lot of times we need to dig into the flux logs which isn't always feasible. Let's discuss this offline - I want to make this a milestone task since a lot of us are finding it hard to pinpoint the error for the failing workflows

**Comment by jonathan-innis** [comment]
Closed with #246
